Classes/BITChannel.m

```diff
@@ -123,39 +128,61 @@ - (BOOL)isQueueBusy {
   return self.channelBlocked;
 }
 
-- (void)persistDataItemQueue {
+- (void)persistDataItemQueue:(char **)string {
   [self invalidateTimer];
-  if (!BITSafeJsonEventsString || strlen(BITSafeJsonEventsString) == 0) {
+  char* jsonStream = NULL;
+  char *newEmptyString = NULL;
+  do {
+    jsonStream = *string;
+    newEmptyString = strdup("");
+    
+    // Reset both, the async-signal-safe buffer and item counter. Swaps jsonStream and BITSafeJsonEventsString.
+    if (OSAtomicCompareAndSwapPtr(jsonStream, newEmptyString, (void*)string)) {
+      self.dataItemCount = 0;
+      break;
+    }
+  } while(true);
+  
+  if (!jsonStream || strlen(jsonStream) == 0) {
+    free(jsonStream);
     return;
   }
   
-  NSData *bundle = [NSData dataWithBytes:BITSafeJsonEventsString length:strlen(BITSafeJsonEventsString)];
+  NSData *bundle = [NSData dataWithBytes:jsonStream length:strlen(jsonStream)];
   [self.persistence persistBundle:bundle];
```

Comment:
Is persistence thread safe too? Meaning can save 2 bundles at the same time without conflicts?